Add DOE CA certs for client authentication

felddy opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

This PR adds the certificates required to authenticate Department of Energy PIVs.  
It generalizes how certificates are downloaded and processed to make it easier to add additional certificate authorities in the future. 

## 💭 Motivation and context ##

<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such
<!-- as "closes" or "resolves" to auto-close them on merge. -->

Changes required to support national labs users who are holders of DOE PIVs. 

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->

<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

Local testing of certificate downloads.
CI test suite.

## 📷 Screenshots (if appropriate) ##

<!-- Remove this section and header if not needed -->

## ✅ Checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
* [x] All new and existing tests pass.

felddy committed:
Add git ignore for certificate directory that can be created in testing.

felddy committed:
Add support for additional certificate authorities.  

Adds the ability to download more than one CA certificate and bundle.
Adds the ability to determine the certificate type and convert it to PEM 
format.
Adds URLs for the Department of Energy.

dav3r commented on files/fetch_user_ca_certs.sh:
```diff
@@ -1,40 +1,74 @@
+  if [[ $url =~ .cer$ ]]; then
+    # Download the certificate and massage into correct format.
+    curl --insecure --silent "${url}" --output "${source_file}"
+    # .cer files contain both PEM and DER formatted certificates.
```
```suggestion
    # .cer files can contain either PEM or DER formatted certificates.
```

dav3r reviewed (approved):
Very nice improvement! 👍
 
My only feedback is a minor change to the wording of one comment to make it a bit clearer.

jsf9k commented:
@felddy - regarding your failed builds, the `ufw` issue is usually resolved by rerunning, possibly multiple times.  I haven't yet been able to pin down the cause of it.  More important is the "invalid cross-device link" error.

felddy committed:
Update files/fetch_user_ca_certs.sh

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

felddy committed:
Change link to copy to avoid cross-mount links.

Some of the molecule images have /tmp as a different mountpoint causing 
a failure.

felddy committed:
Add file package to images that do no have it installed by default.

felddy committed:
Merge branch 'improvement/doe_ca_certs' of github.com:cisagov/ansible-role-openvpn into improvement/doe_ca_certs

felddy committed:
Add file package to Fedora.
